nixos/modules/programs/zsh/zsh.nix

```diff
@@ -162,12 +189,11 @@ in
 
         . /etc/zinputrc
 
-        # history defaults
-        SAVEHIST=2000
-        HISTSIZE=2000
-        HISTFILE=$HOME/.zsh_history
+        export SAVEHIST=${toString cfg.histSize}
+        export HISTSIZE=${toString cfg.histSize}
```

Comment:
Is it possible that zsh truncates the history if SAVEHIST is set to a value like 2000,
even if the user later sets a larger value for example in `interactiveShellInit` below?
Because I sometimes miss old entries.

Reply:
I don't know how zsh behaves in this case. Zsh has the HIST_EXPIRE_DUPS_FIRST option, which removes duplicates when limit is exceeded. I normally put HISTSIZE=1000000 in interactiveShellInit, and it was always enough